src/lib/settings/Configuration.js

```diff
@@ -209,40 +203,35 @@ class Configuration {
 	 * // Updating multiple keys (with arrays):
 	 * Configuration#update(['prefix', 'language'], ['k!', 'es-ES']);
 	 */
-	async update(key, ...args) {
-		if (isObject(key)) {
-			// Overload update(object, GuildResolvable);
-			return this._updateMany(key, args[0] ? this.gateway._resolveGuild(args[0]) : null);
-		}
-
-		// Overload update(string|string[], any|any[], ...any[]);
-		let value = args[0];
-		if (typeof key === 'string') {
-			key = [key];
-			value = [value];
+	update(keys, values, guild, options) {
+		// Overload update(object, GuildResolvable);
+		if (isObject(keys)) {
+			[guild, options] = [values, guild];
+			[keys, values] = objectToTuples(keys);
+		} else if (typeof keys === 'string') {
+			// Overload update(string|string[], any|any[], ...any[]);
+			keys = [keys];
+			values = [values];
+		} else if (!Array.isArray(keys)) {
+			throw new TypeError(`Invalid value. Expected object, string or Array<string>. Got: ${getDeepTypeName(keys)}`);
 		}
 
 		// Overload update(string|string[], any|any[], ConfigurationUpdateOptions);
 		// Overload update(string|string[], any|any[], GuildResolvable, ConfigurationUpdateOptions);
 		// If the third argument is undefined and the second is an object literal, swap the variables.
-		const [guild, options] = typeof args[2] === 'undefined' && args[1] && args[1].constructor.name === 'Object' ?
-			[undefined, args[1]] :
-			[args[1] ? this.gateway._resolveGuild(args[1]) : null, args[2]];
-
-		if (Array.isArray(key)) {
-			if (!Array.isArray(value) || key.length !== value.length) throw new Error(`Expected an array of ${key.length} entries. Got: ${value.length}.`);
-
-			// Create the result with all the promises to resolve
-			const result = { errors: [], updated: [] };
-			const mps = new Array(key.length);
-			for (let i = 0; i < key.length; i++) mps[i] = this._parseSingle(key[i], value[i], guild, options, result);
-			await Promise.all(mps);
-
-			// If at least one key updated, save it to the database
-			if (result.updated.length) await this._save(result);
-			return result;
-		}
-		throw new TypeError(`Invalid value. Expected object, string or Array<string>. Got: ${getDeepTypeName(key)}`);
+		if (typeof options === 'undefined' && guild && guild.constructor.name === 'Object') [guild, options] = [null, guild];
+		if (guild) guild = this.gateway._resolveGuild(guild);
```

Comment:
This should just be changed to use the internal resolver for Klasa. There's no reason for this method to be duplicated for Settings when it should be relatively the same.

Reply:
The problem with this is that at the moment, the Guild's SettingResolver is incredibly slow and requires `await` (`Configuration#update` doesn't use `async` for performance). Plus it takes three arguments and the error message is misleading.